### PR TITLE
Colab materials

### PR DESCRIPTION
**Pull request by ambaha1**
Folder with Colab material to run skai pipeline

### CODE REVIEW

File: colab/colab_utils.py [hunk 1]
```diff
@@ -0,0 +1,1048 @@
...
+import ssl
+import tensorflow as tf
+
+from os import path
+from google.appengine.api import mail
+from google.cloud import monitoring_v3
+from IPython.display import display, HTML, Javascript
+from PIL import Image, ImageDraw, ImageFont
+
+#def launch_pexpect_process(script, arguments,dir_args, use_pexpect):
```
Comment (jzxu) [inline colab/colab_utils.py:34]: Please delete commented out code.

File: colab/colab_utils.py [hunk 2]
```diff
@@ -0,0 +1,1048 @@
+#@title Please run this cell first!
```
Comment (jzxu) [inline colab/colab_utils.py:1]: Is this file the source of the notebook, or a regular python module? If the latter, please delete these "#@" notebook annotations.

File: colab/colab_utils.py [hunk 3]
```diff
@@ -0,0 +1,1048 @@
...
+                   test_acc=test_acc, test_auc=test_auc,
+                   train_epoch=train_epoch,test_epoch=test_epoch,
+                   timestamp=time )
+  return HTML(html)
+
+def timestamp_to_datetime(timestamp):
+  return pd.to_datetime(timestamp)
+
+def update_job_id(job_id):
+    global train_job_id
```
Comment (jzxu) [inline colab/colab_utils.py:666]: We should remove all global variables.

File: colab/colab_utils.py [hunk 4]
```diff
@@ -0,0 +1,1048 @@
...
+import time
+
+import ee
+import folium
+from folium.plugins import HeatMap
+import ipyplot
+import IPython.display
+import pandas as pd
+import pexpect
+import pprint
```
Comment (jzxu) [inline colab/colab_utils.py:20]: Unused imports (that I found): pprint, pytz, smtplib, ssl, google.appengine.api.mail

These should be removed.

File: colab/colab_utils.py [hunk 5]
```diff
@@ -0,0 +1,1048 @@
...
+import os
+import re
+import subprocess
+import time
+
+import ee
+import folium
+from folium.plugins import HeatMap
+import ipyplot
+import IPython.display
```
Comment (jzxu) [inline colab/colab_utils.py:17]: This seems redundant with the "from IPython.display ..." statements below.

File: colab/colab_utils.py [hunk 6]
```diff
@@ -0,0 +1,1048 @@
...
+import pexpect
+import pprint
+import pyproj
+import pytz
+import requests
+import smtplib
+import ssl
+import tensorflow as tf
+
+from os import path
```
Comment (jzxu) [inline colab/colab_utils.py:28]: Every use of path that I found uses the explicit "os.path.<something>" format, so this line can be removed.

File: colab/colab_utils.py (hunk 2 above)
Comment (jzxu) [inline colab/colab_utils.py:1]: Please add a module-level docstring explaining the purpose of this module and its contents. Also, please add a license header.

Comment (jzxu) [inline colab/colab_utils.py:1]: Can you confirm that all functions defined in this file are needed by the notebook? There's quite a lot of code here, a lot of which looks pretty brittle. It'd be good to delete any that isn't actually used.

**Review by jzxu — CHANGES_REQUESTED** [review]
Hi Amine, thanks for creating this PR. In the PR description, please explain that this is the first part of a (I assume) 2-part change, and that there will be a follow-up PR with the actual colab notebook code.

File: colab/colab_utils.py (hunk 1 above)
Comment (ambaha1) [reply colab/colab_utils.py:34]: Thank you,  this will be solve

File: colab/colab_utils.py (hunk 2 above)
Comment (ambaha1) [reply colab/colab_utils.py:1]: This is indeed the regular python module for the specific use of the Colab notebook. The deletions will be done, thank you

File: colab/colab_utils.py (hunk 4 above)
Comment (ambaha1) [reply colab/colab_utils.py:20]: Thank you, those are indeed unused and will be removed.

File: colab/colab_utils.py (hunk 3 above)
Comment (ambaha1) [reply colab/colab_utils.py:666]: Changes will be applied, thank you

File: colab/colab_utils.py (hunk 5 above)
Comment (ambaha1) [reply colab/colab_utils.py:17]: it will be removed, thank you

File: colab/colab_utils.py (hunk 6 above)
Comment (ambaha1) [reply colab/colab_utils.py:28]: I agree, change will be done, thank you

File: colab/colab_utils.py (hunk 2 above)
Comment (ambaha1) [reply colab/colab_utils.py:1]: Module-level docstring and license header will be added, thank you.

Comment (ambaha1) [reply colab/colab_utils.py:1]: Yes I confirm that all those functions are needed to run the Colab notebook, as I decided to reduce to the minimum the amount of code in the notebook. Some pieces of code can probably be even better optimized and refactored for a next future update. (e.g. progress bar, write shell scripts,...).

**Comment by ambaha1** [comment]
Exactly this is part 1 (and was also a test for my first PR to the main branch). Part 2 will come as a seperate PR including the colab notebooks to initialise the skai colab kernel and to run the skai pipeline (the main one)

**Pull request closed without merging**